components/stacks-network/src/chains_coordinator.rs

```diff
@@ -234,14 +254,82 @@ pub async fn start_chains_coordinator(
     let observer_event_tx_moved = observer_event_tx.clone();
     let observer_command_tx_moved = observer_command_tx.clone();
     let ctx_moved = ctx.clone();
+
+    // Load events from snapshot if available
+    let event_pool: Vec<StacksBlockData> = if using_snapshot {
+        let mut events = vec![];
+        let events_cache_path = get_global_snapshot_dir()
+            .join("events_export")
+            .join("events_cache.tsv");
+
+        let mut chain_ctx = StacksChainContext::new(&chainhook_types::StacksNetwork::Devnet);
+        if let Ok(file_content) = fs::read_to_string(&events_cache_path) {
+            for line in file_content.lines() {
+                if line.split('\t').nth(2).unwrap_or("") == "/new_block" {
+                    let maybe_block = standardize_stacks_serialized_block(
+                        &chainhook_sdk::indexer::IndexerConfig {
+                            bitcoin_network: chainhook_types::BitcoinNetwork::Regtest,
+                            stacks_network: chainhook_types::StacksNetwork::Devnet,
+                            bitcoind_rpc_url: config.devnet_config.bitcoin_node_image_url.clone(),
+                            bitcoind_rpc_username: config
+                                .devnet_config
+                                .bitcoin_node_username
+                                .clone(),
+                            bitcoind_rpc_password: config
+                                .devnet_config
+                                .bitcoin_node_password
+                                .clone(),
+                            bitcoin_block_signaling: BitcoinBlockSignaling::Stacks(
+                                StacksNodeConfig {
+                                    rpc_url: config.devnet_config.stacks_node_image_url.clone(),
+                                    ingestion_port: 3999,
+                                },
+                            ),
+                        },
+                        line.split('\t').nth(3).unwrap_or(""),
+                        &mut chain_ctx,
+                        &ctx,
+                    );
+                    match maybe_block {
+                        Ok(block) => {
+                            events.push(block);
+                        }
+                        Err(e) => {
+                            let _ =
+                                devnet_event_tx.send(DevnetEvent::debug(format!("Error: {}", e)));
+                        }
+                    }
+                }
+            }
+        }
+
+        let _ = observer_event_tx.send(ObserverEvent::StacksChainEvent((
+            StacksChainEvent::ChainUpdatedWithBlocks(StacksChainUpdatedWithBlocksData {
+                new_blocks: events
+                    .clone()
+                    .iter()
+                    .map(|event| StacksBlockUpdate::new(event.clone()))
+                    .collect::<Vec<_>>(),
```

Comment:
This can be cleaned up and cloning `events` is not necessary:

```suggestion
                new_blocks: events
                    .iter()
                    .cloned()
                    .map(StacksBlockUpdate::new)
                    .collect(),
```